wolfssl/src/ssl.rs

```diff
@@ -714,6 +714,29 @@ impl<IOCB: IOCallbacks> Session<IOCB> {
         }
     }
 
+    /// Invokes [`wolfSSL_dtls13_use_quick_timeout`][0]
+    ///
+    /// [0]: https://www.wolfssl.com/documentation/manuals/wolfssl/ssl_8h.html#function-wolfssl_dtls13_use_quick_timeout
+    pub fn dtls13_use_quick_timeout(&self) -> bool {
+        if !self.is_dtls() {
+            log::debug!("Session is not configured for DTLS");
+            return false;
+        }
+
+        let ssl = self.ssl.lock();
+        // SAFETY: [`wolfSSL_dtls13_use_quick_timeout`][0] ([also][1]) expects a valid pointer to `WOLFSSL`. Per the
+        // [Library design][2] access is synchronized via the containing [`Mutex`]
+        //
+        // [0]: https://www.wolfssl.com/documentation/manuals/wolfssl/ssl_8h.html#function-wolfssl_dtls13_use_quick_timeout
+        // [1]: https://www.wolfssl.com/doxygen/ssl_8h.html#a61f3b53cb0397dd1debc8b8daaa490c2
+        // [2]: https://www.wolfssl.com/documentation/manuals/wolfssl/chapter09.html#thread-safety
+        match unsafe { wolfssl_sys::wolfSSL_dtls13_use_quick_timeout(ssl.as_ptr()) } {
+            0 => false,
+            1 => true,
+            e => unreachable!("{e:?}"),
```

Comment:
Docs say:

> Return: true if the application code should setup a quicker timeout

In C true is any non-zero value not specifically 1. So I think:
```rust
match ... {
    0 => false
    _ => true
}
```

might be closer to what the API docs say?

Or equivalently a `!= 0` check. Bit of a shame there is no `From<c_int> for bool`, but it is what it is.

Reply:
All the existing code follows this convention of converting <c_int> to bool. 
Think its better to be consistent. We can followup with a Jira/PR, if we want to update all.

Reply:
Other cases e.g. https://www.wolfssl.com/documentation/manuals/wolfssl/group__TLS.html#function-wolfssl_is_init_finished or https://www.wolfssl.com/documentation/manuals/wolfssl/ssl_8h.html#function-wolfssl_dtls the docs do specifically say 0 and 1. This one doesn't.

Reply:
I think the other way. 

In C, true is actually `1` if we use stdbool. Using any value other than 0 for true is a convention or hack.

https://pubs.opengroup.org/onlinepubs/009695399/basedefs/stdbool.h.html
https://android.googlesource.com/platform/prebuilts/gcc/linux-x86/arm/arm-eabi-4.8/+/4cedbb07c5475ce9a5fc9f2d9645c00292e4be72/lib/gcc/arm-eabi/4.8/include/stdbool.h#34

If we go by this logic, think the changes are relevant. 
Also it is consistent :man_shrugging:

Reply:
There is no stdbool involved here:
```
WOLFSSL_API int wolfSSL_dtls13_use_quick_timeout(WOLFSSL* ssl);
```

The treatment of an integer type a boolean context in C, where "non-zero is true", is neither a hack nor merely a convention, it's part of the definition of the language.

WRT consistency, the wolfssl API is inconsistent here it seems, so it follows that we will be too.

Reply:
Ack! Most C library function return true(nonzero) value in general
So updated the code as per above suggestion